### ISSUE TITLE
[v1.3.0] Release Tracker

### ISSUE THREAD

**Issue by soumith** [pytorch/pytorch#27011]
The branch v1.3.0 has been cut.

If you need any particular patches onto this branch, please comment below and send a PR to v1.3.0 branch (instead of master).

---

Current PRs open against the `v1.3.0` branch: https://github.com/pytorch/pytorch/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+base%3Av1.3.0

**Comment by ljk53** [comment]
#27014 - fix a recent mobile build size regression

**Comment by vishwakftw** [comment]
https://github.com/pytorch/pytorch/pull/27018 - removes an incorrect note from the doc strings of `cholesky_solve` and `triangular_solve`.

---
Status: PR open against v1.3.0 at #27018

**Comment by ailzhang** [comment]
Serialize XLA Tensor 
---
Status: PR open against v1.3.0 at https://github.com/pytorch/pytorch/pull/27042

**Comment by ngimel** [comment]
make cudnn rnn respect current stream

Status: PR open against v1.3.0 #27044

**Comment by raghuramank100** [comment]
Per channel quantization baseline for post training support
https://github.com/pytorch/pytorch/pull/26516

**Comment by raghuramank100** [comment]
Per channel fake-quant
https://github.com/pytorch/pytorch/pull/26623

**Comment by raghuramank100** [comment]
Quantization aware training: Freeze batch norm support
https://github.com/pytorch/pytorch/pull/26624

**Comment by raghuramank100** [comment]
Emulate weight and activation only quant with fake quant
https://github.com/pytorch/pytorch/pull/26625

**Comment by raghuramank100** [comment]
Default observer and fake-quant for backends
https://github.com/pytorch/pytorch/pull/26627

**Comment by ifedan** [comment]
Fix test_overwrite_module_params_on_conversion_cpu_cuda  after comparison ops type promotion was introduced
https://github.com/pytorch/pytorch/pull/27066

**Comment by jamesr66a** [comment]
Stack of 2:
https://github.com/pytorch/pytorch/pull/27075
https://github.com/pytorch/pytorch/pull/27076

**Comment by ZolotukhinM** [comment]
#27012 fix for a race condition in `torch::jit::Function::optimized_graph()`

**Comment by wanchaol** [comment]
https://github.com/pytorch/pytorch/pull/27079 add autograd ops serialization for TorchScript

**Comment by houseroad** [comment]
https://github.com/pytorch/pytorch/pull/27049 bump up the onnx exporter producer version

**Comment by zou3519** [comment]
https://github.com/pytorch/pytorch/pull/27084: turn on all named tensor tests on v1.3.0 branch so that whoever is testing binaries will see the results of those tests.

---
Status: PR open against v1.3.0 at https://github.com/pytorch/pytorch/pull/27084

**Comment by pietern** [comment]
#27087 (PR against the 1.3.0 branch) bumps Gloo with a torch.distributed for macOS bugfix.

**Comment by IvanKobzarev** [comment]
https://github.com/pytorch/pytorch/pull/27090 adding to android api  method Module#destroy that is used by demo application and described in tutorials and docs for 1.3

**Comment by IvanKobzarev** [comment]
https://github.com/pytorch/pytorch/pull/27111 - android api org.pytorch.Module#destroy method added

Status: PR open against v1.3.0

**Comment by eellison** [comment]
https://github.com/pytorch/pytorch/pull/27112 [jit] Fix `toIValue` dict iteration 
@driazati 

Open against 1.3.0

**Comment by peterjc123** [comment]
#27120 Fix Windows CI for 1.3.0

Status: PR open against v1.3.0

**Comment by peterjc123** [comment]
#27069 Fixed seek offset size to 64bit. 

Status: PR open against v1.3.0

**Comment by xta0** [comment]
#27145 Update the link in README.md

Status: PR open against v1.3.0

**Comment by xuhdev** [comment]
#26993 Fix a recent regression that lost some of our patches to Sleef

Status: PR open against v1.3.0

**Comment by ezyang** [comment]
#27216 Make static dispatch turn off variable before entering the kernel

Status: PR open against v1.3.0

**Comment by eellison** [comment]
https://github.com/pytorch/pytorch/pull/27226: make class types callable (#26743) 

Status: PR open against v1.3.0

**Comment by houseroad** [comment]
https://github.com/pytorch/pytorch/pull/27233: Fix the ONNX interpolate export bug

Status: PR open against v1.3.0

**Comment by raghuramank100** [comment]
Enhancements to fuse module support for quantization
https://github.com/pytorch/pytorch/pull/26457

**Comment by raghuramank100** [comment]
Support for add_relu functional module
https://github.com/pytorch/pytorch/pull/26612

**Comment by raghuramank100** [comment]
Handle un-initialized values in histogram observer
https://github.com/pytorch/pytorch/pull/27151

**Comment by raghuramank100** [comment]
Add control for observers in Fake-quantize module
https://github.com/pytorch/pytorch/pull/27113

**Comment by izdeby** [comment]
Enabled comparison ops with named tensors 
https://github.com/pytorch/pytorch/pull/27249

**Comment by izdeby** [comment]
Fixed Error message for tensor.align_to
https://github.com/pytorch/pytorch/pull/27250

**Comment by IvanKobzarev** [comment]
pytorch android version change to 1.3.0 in gradle.properties
PR to v1.3.0 https://github.com/pytorch/pytorch/pull/27275

**Comment by jamesr66a** [comment]
@soumith we're consolidating all the quantization commits into this branch/PR: https://github.com/pytorch/pytorch/pull/27303

**Comment by Krovatkin** [comment]
@soumith fixing a segfault when a user uses unsupported list comprehension arguments https://github.com/pytorch/pytorch/pull/27305

**Comment by ljk53** [comment]
#27321 - fix USE_STATIC_DISPATCH for mobile (replacement of #27216)

Status: PR open against v1.3.0 at #27321

**Comment by ZolotukhinM** [comment]
> #27012 fix for a race condition in `torch::jit::Function::optimized_graph()`

This is the corresponding Pull Request to `v1.3.0` branch: #27323.

**Comment by soumith** [comment]
@jamesr66a rebase over current v1.3.0 when you get a chance, CI should be fixed-ish

@Krovatkin you closed the PR, do you not want it in v1.3.0 anymore?

**Comment by pietern** [comment]
Please double check that the macOS wheels are built with torch.distributed support. Thanks!

**Comment by zou3519** [comment]
https://github.com/pytorch/pytorch/pull/27367 make `align_to` method only. Exists in master as https://github.com/pytorch/pytorch/pull/27304

---
Status: PR open against v1.3.0 at #27367, waiting for CI 💛

**Comment by soumith** [comment]
https://github.com/pytorch/pytorch/pull/26980 Make nonzero non-differentiable as it should be

---
Status: merged via 3c8ce2a57e

**Comment by Krovatkin** [comment]
@soumith 

> @Krovatkin you closed the PR, do you not want it in v1.3.0 anymore?


I'm actually very confused. As far as I can tell my fix is **already** in `v1.3.0` : https://github.com/pytorch/pytorch/blob/v1.0.3/torch/csrc/jit/script/compiler.cpp#L1140

I guess, we have pulled quite a few changes from `master` into `v1.3.0`?

**Comment by Krovatkin** [comment]
@soumith 

https://github.com/pytorch/pytorch/pull/27398

**Comment by IvanKobzarev** [comment]
PR https://github.com/pytorch/pytorch/pull/27407 android torchvision - Add methods to write image tensor content to buffer, need for demo app
status: PR to `v1.3.0`

**Comment by AshkanAliabadi** [comment]
PR #27341 - Modify PyTorch's integration of NNPACK to use a unified underlying thread pool implementation.

**Comment by gottbrath** [comment]
Pr [27414](https://github.com/pytorch/pytorch/pull/27414) with docstring only formatting changes.

**Comment by t-vi** [comment]
Ideally, we would have #27039 (CuDNN CTC loss) and #27460 (native CTC loss for long target lengths).
The CuDNN one is very ugly, though (writing into the memory for the CuDNN loss descriptor).

**Comment by ailzhang** [comment]
Make not specifying `align_corners` a hard error. 

---
status: https://github.com/pytorch/pytorch/pull/27468 opened for 1.3.0

**Comment by driazati** [comment]
#27386: [jit] Reduce special casing around 'training'

Status: #27386 opened against v1.3.0

**Comment by pbelevich** [comment]
#27480 #include <stdexcept> into flat_hash_map.h to fix #27266

**Comment by nairbv** [comment]
opened https://github.com/pytorch/pytorch/pull/27501 for doc improvements on torch.add including type promotion information.

**Comment by AshkanAliabadi** [comment]
https://github.com/pytorch/pytorch/pull/27496 against v1.3.0

**Comment by soumith** [comment]
@dreiss posted the following android picks: https://github.com/pytorch/pytorch/pull/27516

---
Status: waiting on CI 💛

**Comment by jamesr66a** [comment]
@zafartahirov has https://github.com/pytorch/pytorch/pull/27511

**Comment by t-kuha** [comment]
Please include #27533 into v1.3.0.

**Comment by driazati** [comment]
#27545 

Waiting on CI, landing in master in #27514

**Comment by ljk53** [comment]
#27547

Waiting on CI

**Comment by ailzhang** [comment]
https://github.com/pytorch/pytorch/pull/27557 

----
fix jit test failures

**Comment by zou3519** [comment]
Documentation for named tensors: https://github.com/pytorch/pytorch/pull/27573. Corresponding PR to master: https://github.com/pytorch/pytorch/pull/27173

---
Status: Waiting on CI 💛

**Comment by t-kuha** [comment]
Please include #27587.
(Fixed target branch to v1.3.0)

**Comment by elbamos** [comment]
Please note issue  #27627. I apologize for not reporting it until now - I needed to clear out some issues that were impeding compilation of 1.2 to confirm the issue before reporting it.

**Comment by zou3519** [comment]
Docs CI build job https://github.com/pytorch/pytorch/pull/27638

---
Status: Not critical to get into the release. waiting for CI  💛

**Comment by jlin27** [comment]
Add Javadoc for v1.3.0: https://github.com/pytorch/pytorch/pull/27656

Quant doc changes: https://github.com/pytorch/pytorch/pull/27640

**Comment by gottbrath** [comment]
If it isn't too late please merge https://github.com/pytorch/pytorch/pull/27574

**Comment by jlin27** [comment]
Quant and torch changes:
https://github.com/pytorch/pytorch/pull/27701

https://github.com/pytorch/pytorch/pull/27702

https://github.com/pytorch/pytorch/pull/27639
---------------------------------
(Just as reference. Per Dima's PR to master: https://github.com/pytorch/pytorch/pull/27677)

**Comment by fightthepower** [comment]
I can't install pytorch 1.3 in my system currently I have 1.2. This is the error 

 ```
pip3 install --upgrade torch==1.3.0+cpu torchvision==0.4.1+cpu -f https://download.pytorch.org/whl/torch_stable.html

Looking in links: https://download.pytorch.org/whl/torch_stable.html
Collecting torch==1.3.0+cpu
  ERROR: Could not find a version that satisfies the requirement torch==1.3.0+cpu (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2, 0.1.6.post17, 0.1.6.post20, 0.3.0.post4, 0.3.1, 0.4.0, 0.4.1, 1.0.0, 1.0.1, 1.0.1.post2, 1.1.0, 1.2.0, 1.2.0+cpu, 1.2.0+cu92, 1.3.0, 1.3.0+cu100, 1.3.0+cu92)
ERROR: No matching distribution found for torch==1.3.0+cpu

```

**Comment by wizardk** [comment]
#26948 : Please optimizing the CPU performance in this branch.

**Comment by soumith** [comment]
@fightthepower thanks for reporting. fixed now.
